include/maths/time_series/CTimeSeriesDecompositionDetail.h

```diff
@@ -91,7 +97,8 @@ class MATHS_TIME_SERIES_EXPORT CTimeSeriesDecompositionDetail
                   CTimeSeriesDecomposition& decomposition,
                   const TMakePredictor& makePredictor,
                   const TMakeFilteredPredictor& makeSeasonalityTestPreconditioner,
-                  const TMakeTestForSeasonality& makeTestForSeasonality);
+                  const TMakeTestForSeasonality& makeTestForSeasonality,
+                  const TModelAllocator& allocator = common::CModelAllocatorStub());
```

Comment:
Maybe we can have a static instance to avoid repeatedly creating temporary objects on call. For example you could have an instance on `CModelAllocatorStub`.

Reply:
Done